pkg/executor/push.go

```diff
@@ -114,7 +115,15 @@ func CheckPushPermissions(opts *config.KanikoOptions) error {
 		if checked[destRef.Context().RepositoryStr()] {
 			continue
 		}
+		if strings.Contains(destRef.RegistryStr(), "gcr.io") {
```

Comment:
Comments explaining why you are eating this would be super-helpful for future devs.

Reply:
sorry, not fully following, would a comment like this help?
"Checking if user provided .docker/config.json is present"
the reason I eat the error is because I only care if the file doesn't exist.

Reply:
Not the what, why are you checking for config.json?  I have a pretty good idea why because of the bug, but it will not be obvious why you are doing that for future devs.
```
// Checking for existence of docker.config as it's normally required for authenticated registries. However, if it doesn't exist, we don't want to error out since it's not required for public (non-authenticated) registries.
```

Reply:
ah yeah that makes a lot more sense. I somehow thought you were fixed on the `err` variable lol. I added comments to explain the why part.